easybuild/easyconfigs/z/zfp/zfp-0.5.5-GCCcore-10.2.0.eb

```diff
@@ -0,0 +1,33 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Denis Kristak
+easyblock = 'MakeCp'
```

Comment:
It seems like `zfp` supports building with just `make`, or via `cmake` (because, you know, why not...).

Although I'm not a big fan of CMake, it's probably better to go that way instead, since it gives us more control over compiler options, etc, and it hopefully also avoids the need for "manually" installing stuff via `files_to_copy`, and we can just use `make install` instead.

I'll take a quick look at the myself, so we can get this merged and proceed with #14247 ...

Reply:
And it actually supports the `configopts`, which.. I suspect does absolutely nothing in MakeCp?

Reply:
Indeed, `configopts` doesn't make sense with `MakeCp`.

I just spent an hour trying to get `zfp` built with CMake, and it's living up to its reputation, again...

Problems I ran into before I gave up:

* The build was failing while linking the `zfp` binary due to an undefined reference to `log10`, basically because `-lm` wasn't being used. There's some logic in `CMakeLists.txt` to check whether `-lm` is required or not, but it only checks for compilation, not linking. Even after replacing `check_c_source_compiles` with `check_c_source_runs`, it still concludes that `-lm` isn't needed at all, and then fails to link `zfp` because `-lm` isn't used.
  I was able to dance around that by force-feeding CMake `-lm` via `-DCMAKE_EXE_LINKER_FLAGS='-lm'`.
* After fixing the `zfp` linking issue, the sanity check was failing because `bin/zfp` wasn't being installed via `make install`, and it was building `libzfp.so` by default (while using the `Makefile` it builds `libzfp.a` by default).
  I couldn't figure out why CMake in all its wisdom as building the `zfp` utility to then not install it at all, so I gave up.

Conclusion: if you get the option to either just run `make` or build through `cmake`, go with the former.